fix: Improve ergonomics for the Decimal type

Divyanshu-s13 opened the pull request:
## What's Changed

Added a new decimal.ts module that provides high-level utilities for working with Decimal128 and Decimal256 values in JavaScript. These utilities make decimal handling more ergonomic by removing the need for low-level bit manipulation and aligning behavior with Arrow C++ semantics. The module is exported through the util namespace and integrates cleanly without introducing breaking changes.

Closes #81.

Commit:
Fix #81: Improve ergonomics for the Decimal type in JavaScript bindings

domoritz reviewed (changes requested):
Thanks for the pull request. This definitely needs tests.

Also, does this address https://github.com/apache/arrow-js/issues/100 and https://github.com/apache/arrow-js/issues/77 in some way?

Commit:
test: add decimal utilities test suite

- Comprehensive tests for isNegativeDecimal, negateDecimal
- Tests for toDecimalString and toDecimalNumber conversions
- Tests for fromDecimalString parsing with roundtrips
- Integration tests for negation and edge cases

Commit:
fix:CI failure

Commit:
fix: resolve ESLint errors in decimal.ts

- Change 'let n' to 'const n' (prefer-const)
- Use ternary for magnitude assignment (unicorn/prefer-ternary)
- Use ternary for result composition (unicorn/prefer-ternary)

Divyanshu-s13 commented:
> Could you check lint failures?
> 
> https://github.com/apache/arrow-js/actions/runs/19644239168/job/56300291328?pr=341#step:5:8
> 
> ```
> /home/runner/work/arrow-js/arrow-js/src/util/decimal.ts
> Error:    84:9  error  'n' is never reassigned. Use 'const' instead                 prefer-const
> Error:    89:5  error  This `if` statement can be replaced by a ternary expression  unicorn/prefer-ternary
> Error:   128:5  error  This `if` statement can be replaced by a ternary expression  unicorn/prefer-ternary
> ```
> 
> FYI: You can run CI on your fork by enabling GitHub Actions on your fork.

I have fix this can you merge it.

domoritz reviewed (commented):
Thanks for making these utilities. I wonder, can we also use some of these utilities in other tests that already exist?
